src/Refactoring-Core/RBTemporaryToInstanceVariableRefactoring.class.st

```diff
@@ -143,6 +135,31 @@ RBTemporaryToInstanceVariableRefactoring >> isTemporaryVariableNameValid [
 		self refactoringError: temporaryVariableName , ' is a block parameter' ]
 ]
 
+{ #category : 'as yet unclassified' }
+RBTemporaryToInstanceVariableRefactoring >> preconditionNoMultipleTempOccurences [
+
+	^ ReMultipleMethodsDontReferToTempVarsCondition name: temporaryVariableName class: class selector: selector 
+]
+
+{ #category : 'as yet unclassified' }
+RBTemporaryToInstanceVariableRefactoring >> preconditionNoReadBeforeWritten [
+
+	^ ReVariablesNotReadBeforeWrittenCondition new
+		  subtree: parseTree;
+		  variables: temporaryVariableName;
+		  checkForTemporaryVariables: true 
+]
+
+{ #category : 'as yet unclassified' }
+RBTemporaryToInstanceVariableRefactoring >> preconditionNoSubclassDefinesVar [
+
+	^ RBCondition
+		  withBlock: [ (class allSubclasses anySatisfy: [ :cls | cls definesInstanceVariable: temporaryVariableName asString ]) not ]
```

Comment:
In a second PR we should remove this pop up!

Reply:
> In a second PR we should remove this pop up!

Yes, I suggested Alexis to leave it like this for the moment. 
After merging [this PR](https://github.com/pharo-project/pharo/pull/18738) refactoring the method will be way easier